File: python/packages/package-unit/src/package_unit/services/embedding_service.py
from __future__ import annotations

from typing import List, Dict

from package_unit.utils.http import post_json


class EmbeddingService:
    @staticmethod
    def get_embedding(text: str, auth_headers: Dict[str, str]) -> List[float]:
        resp = post_json(
            "llm/predict",
            json={
                "uri": "/v1/embeddings",
                "provider": "chatgpt",
                "model": "text-embedding-3-small",
                "payload": {"input": text},
            },
            headers={"Content-Type": "application/json", **(auth_headers or {})},
        )
        try:
            data = resp.json()
            return data.get("data", [{}])[0].get("embedding", [])
        except Exception:
            return []

File: node/packages/content/src/services/EmbeddingService.ts
export class EmbeddingService {
  static async getEmbedding(text: string, authHeaders: Record<string, string>): Promise<number[]> {
    const response = await fetch(`${process.env.API_URL}/llm/predict`, {
      method: "POST",
      headers: { "Content-Type": "application/json", ...authHeaders },
      body: JSON.stringify({
        uri: "/v1/embeddings",
        provider: "chatgpt",
        model: "text-embedding-3-small",
        payload: { input: text }
      })
    });

    const data = await response.json();
    return data?.data?.[0]?.embedding ?? [];
  }
}
